pkg/cmd/provisioning/nutanix/create_shared_secrets_test.go

```diff
@@ -0,0 +1,199 @@
+package nutanix
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	testCredReqDirPrefix     = "nutanix_test_dir_credreq"
+	testTargetDirPrefix      = "nutanix_test_dir_target"
+	testCredentialsDirPrefix = "nutanix_test_dir_credentials"
+
+	credReqTemplate = `---
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: %s
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: %s
+  secretRef:
+    name: %s
+    namespace: %s`
+
+	basicAuthCredentialsTemplate = `
+credentials:
+- type: basic_auth
+  data:
+    prismCentral:
+      username: %s
+      password: %s`
+)
+
+func TestCreateSharedSecrets(t *testing.T) {
+	tests := []struct {
+		name        string
+		setup       func(*testing.T) (credReqDir, targetDir, credentialsSourceFilepath string)
+		verify      func(*testing.T, string)
+		expectedErr string
+	}{
+		{
+			name: "No CredentialsRequest manifests in directory",
+			setup: func(t *testing.T) (credReqDir, targetDir, credentialsSourceFilepath string) {
+				credReqDir, err := ioutil.TempDir(os.TempDir(), testCredReqDirPrefix)
+				require.NoError(t, err, "Failed to create temp directory for credentials requests")
+
+				targetDir, err = ioutil.TempDir(os.TempDir(), testTargetDirPrefix)
+				require.NoError(t, err, "Failed to create temp directory for credentials requests")
+
+				credentialsDir, err := ioutil.TempDir(os.TempDir(), testCredentialsDirPrefix)
+				require.NoError(t, err, "Failed to create temp directory for credentials")
+				credentialsSourceFilepath = testBasicAuthCredentials(t, "username", "password", credentialsDir)
+				return
+			},
```

Comment:
`setup` function does not return anything, I am curious how this works. Is it based on variable names?

Reply:
Indeed, these are named return parameters. The variables which will be returned are defined in the function signature.

Reply:
thanks! learned something new :)